p2p/src/network/noise/p2p_network_noise_reducer.rs

```diff
@@ -188,55 +190,72 @@ impl P2pNetworkNoiseState {
                     } => {
                         let aead = ChaCha20Poly1305::new(&send_key.0.into());
                         let chunk_max_size = u16::MAX as usize - 19;
-                        let chunks = data
-                            .chunks(chunk_max_size)
-                            .map(|data| {
-                                let mut chunk = Vec::with_capacity(18 + data.len());
-                                chunk.extend_from_slice(&((data.len() + 16) as u16).to_be_bytes());
-                                chunk.extend_from_slice(data);
+                        let mut chunks = vec![];
+
+                        for data in data.chunks(chunk_max_size) {
+                            let mut chunk = Vec::with_capacity(18 + data.len());
+                            chunk.extend_from_slice(&((data.len() + 16) as u16).to_be_bytes());
+                            chunk.extend_from_slice(data);
+
+                            let mut nonce = GenericArray::default();
```

Comment:
Could you add an explicit type signature for `nonce`? It is not obvious to figure out what array size the compiler has inferred.